src/compile.rs

```diff
@@ -499,6 +499,7 @@ pub(crate) fn compile_inner(inner_re: &str, options: &RegexOptions) -> Result<Ra
 
     let re = RaBuilder::new()
         .configure(config)
+        .syntax(options.syntaxc)
```

Comment:
Is this the only place where we actually construct a case insensitive matcher at the moment?

Reply:
To the best of my knowledge and search, I believe this is the only place that we need this.

Reply:
@robinst Just wanted to bump this, was there anything you feel like is missing currently for `lowercase` specific building? 

I'd like to iterate over the rest of the builder functionality as a future works.

Reply:
Hey @jonperry-dev, thanks for your patience. Yeah, let's go ahead with this. Can you rebase the PR? The builds seem to have expired, sorry about that.

Reply:
No worries at all! Thanks for getting back to me. 

Just updated, should be good to go.

I hope everything is well! 🙂

Reply:
@robinst I think its now waiting for maintainer approval on CI. I did a few more changes: 
- Cleaned up the code
- had a `cargo fmt` change
- moved tests to its own file (plan on making more extensions to the options)